components/locid/src/macros.rs

```diff
@@ -1,342 +0,0 @@
-// This file is part of ICU4X. For terms of use, please see the file
-// called LICENSE at the top level of the ICU4X source tree
-// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
-
-/// A macro allowing for compile-time construction of valid [`Language`] subtag.
```

Comment:
Nit (optional): I'd prefer keeping the macros in one place, because

1. I think they are more similar to each other than to the class they are constructing
2. @yzhang1994 has an open PR where this change causes a needless merge conflict
3. In general, I don't like moving code around because it makes `git blame` not work as well, unless there's a compelling reason, and I don't see the compelling reason here